### PR TITLE
Feature stateless

### LINKED ISSUE #1043: [Field] Stateless component in a <Field /> raises a warning.
I am using an inline stateless component for the `DisplayComponent` prop of `fieldFor` and I am prompted with this warning : 
`Warning: Stateless function components cannot be given refs (See ref "display" in DisplayComponent created by null). Attempts to access this ref will fail.`
The message is pretty much self-explanatory.

### PR DESCRIPTION
**Pull request by pierr**
The goal of this PR is to remove all warnings due to  pure function use

- [x] All `field` methods `fieldFor`, displayFor`, selectFor`, autoCompleteFor`, `autoCompleteSelectFor`
- [x] All `list` related methods `TimeLine`, 
- [x] Add a util method to know is a component is pure or not
- [x] Add a util method to add a ref to a props object only when there is a 

It fixes all the warnings with `ref` problems.
It fixes #1043

### CODE REVIEW

File: src/utils/is-react-class-component.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+// is a component a react class.
+export const isReactClassComponent = ComponentToTest => {
+ const prototype = ComponentToTest.prototype;
+  if (!prototype) {
+    return false;
+  }
+  return typeof prototype.render === 'function';
+}
+
```
Comment (pierr) [inline src/utils/is-react-class-component.js]: Maybe the doc is missing here.

File: src/utils/is-react-class-component.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+//This function aims to test if a component is a
+
+// is a component a react class.
+export const isReactClassComponent = ComponentToTest => {
+ const prototype = ComponentToTest.prototype;
+  if (!prototype) {
```
Comment (Bernardstanislas) [inline src/utils/is-react-class-component.js]: no prototype for functions ? I didn't know

File: src/utils/is-react-class-component.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    return false;
+  }
+  return typeof prototype.render === 'function';
+}
+
+export const addRefToPropsIfNotPure = (Component, props, ref) => (isReactClassComponent(Component) ? {...props, ref} : props);
+
+
+export const LIST = 'list';
```
Comment (Bernardstanislas) [inline src/utils/is-react-class-component.js]: maybe these should be located in each component ? I might be wrong

File: src/common/field/mixin/built-in-components.js [hunk 4]
```diff
@@ -108,15 +111,16 @@ const fieldBuiltInComponentsMixin = {
         const {name: id, placeholder} = this.props;
         const {value, error} = this.state;
         const {onInputChange: onChange} = this;
-        const inputBuildedProps = assign({}, this.props, {
+        const inputBuildedProps = {
```
Comment (Bernardstanislas) [inline src/common/field/mixin/built-in-components.js]: `inputBuiltProps`

Comment (pierr) [inline src/common/field/mixin/built-in-components.js]: @Bernardstanislas You can do a PR for that.

**Pull request merged**